scylla/src/transport/connection.rs

```diff
@@ -1292,6 +1253,69 @@ pub async fn open_named_connection(
     Ok((connection, error_receiver))
 }
 
+async fn start_authenticate(
+    connection: &mut Connection,
+    authenticate: &Authenticate,
+) -> Result<(), QueryError> {
+    let authenticator: Authenticator = match &authenticate.authenticator_name as &str {
+        "AllowAllAuthenticator" => AllowAllAuthenticator,
+        "PasswordAuthenticator" => PasswordAuthenticator,
+        "org.apache.cassandra.auth.PasswordAuthenticator" => CassandraPasswordAuthenticator,
+        "org.apache.cassandra.auth.AllowAllAuthenticator" => CassandraAllowAllAuthenticator,
+        "com.scylladb.auth.TransitionalAuthenticator" => ScyllaTransitionalAuthenticator,
+        _ => unimplemented!(
+            "Authenticator not supported, {}",
+            authenticate.authenticator_name
+        ),
+    };
```

Comment:
Let's get rid of this code and switch to using just `String`/`&str` instead of an enum to represent the authenticator name. The `Authenticator` itself should verify in `start_authentication_session` that it recognizes and supports the authentication kind.